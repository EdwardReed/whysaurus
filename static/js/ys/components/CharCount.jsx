import React from 'react'
import PropTypes from 'prop-types'

export class CharCount extends React.Component {

  constructor(props) {
    super(props);
    this.updateCharCountOnChange = this.updateCharCountOnChange.bind(this);
    this.state = {
      charsLeft: props.countedValue ? (props.maxChars - props.countedValue.length) : props.maxChars
    };
  }

  updateCharCountOnChange(text) {
    this.setState({
      charsLeft: this.props.maxChars - text.length
    });
  }

  render(){
    return (
      this.props.render({charsLeft: this.state.charsLeft, countedFieldOnChange: this.updateCharCountOnChange})
    )
  }
}

CharCount.propTypes = {
  render: PropTypes.func.isRequired,
  maxChars: PropTypes.number.isRequired
}

export default CharCount
