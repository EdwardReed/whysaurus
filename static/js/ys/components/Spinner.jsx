import React from 'react'

export default class Spinner extends React.Component {

  // TO DO: add options to change color
  render(){
    return <img className="spinner" src="/static/img/Reload-1s-200px_blue_transp.svg"/>
  }
}
      
