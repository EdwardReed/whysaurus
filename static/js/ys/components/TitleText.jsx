import React from 'react'
import PropTypes from 'prop-types'
import { Form, Text, Field } from 'react-form'
import CharCount from './CharCount'
import * as validations from '../validations'
import * as formUtils from '../form_utils.js'

const TitleText = props => (
  <Field field={props.field}>
    {fieldApi => {
      // field is here to strip that property out of `rest`
      const { field, ...rest } = props

      const { value, error, warning, success, setValue, setTouched } = fieldApi

      return (
        <div className="titleField">
          <CharCount maxChars={validations.titleMaxCharacterCount} render={({charsLeft, countedFieldOnChange}) => (
            <div>
              <Text onChange={countedFieldOnChange} field="title" {...rest}/>
              <p classes={charsLeft && charsLeft < 0 ? 'overMaxChars' : ''}>{charsLeft}</p>
              <p>{error}</p>
            </div>
              )}/>
        </div>
      )
    }}
  </Field>
);

TitleText.propTypes = {
}

export default TitleText
