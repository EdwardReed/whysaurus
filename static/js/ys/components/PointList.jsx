import 'babel-polyfill';
import React from 'react';
import ReactDOM from 'react-dom';
import gql from 'graphql-tag';
import { graphql } from 'react-apollo';
import { withRouter } from 'react-router';
import InfiniteScroll from 'react-infinite-scroll-component';
import PropTypes from 'prop-types';


import Spinner from './Spinner'
import { GetPoint } from '../schema';
import { PointCard } from './Point';
import config from '../config'

class PointListComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      expandedIndex: this.urlExpandedIndex(),
      hideIrrelevant: true
    }
  }

  get parentPoint() {
    return this.props.parentPoint;
  }

  isPointExpanded = (point) => {
    return !!this.state.expandedIndex[point.url]
  }

  expandedIndex2Param = (index) => {
    console.log(index)
    return JSON.stringify(Object.keys(index).filter(key => index[key]))
  }

  expandedParam2Index = (param) => {
    try {
      return param ? JSON.parse(param).reduce((i, k) => {i[k] = true; return i}, {}) : {}
    } catch (err) {
      console.log("Error parsing URL expanded index:")
      console.log(param)
      console.log(err)
      return {}
    }
  }

  urlExpandedIndex = (urlSearchParams) => {
    let params = urlSearchParams || new URLSearchParams(this.props.location.search)
    return this.expandedParam2Index(params.get("expanded"))
  }

  updateURLExpandedIndex = (urlSearchParams, updatedIndex) => {
    let params = urlSearchParams || new URLSearchParams(this.props.location.search)
    if (Object.entries(updatedIndex).length > 0) {
      params.set("expanded", this.expandedIndex2Param(updatedIndex))
    } else {
      params.delete("expanded")
    }
    const { location, history } = this.props
    history.push({path: location.pathname, search: params.toString()})
  }

  modifyStateAndURL = (point, indexModifier) => {
    let params = new URLSearchParams(this.props.location.search);
    let index = this.urlExpandedIndex()
    indexModifier(index)
    this.setState({expandedIndex: index})
    this.updateURLExpandedIndex(params, index)
  }

  handleSeeEvidence = (point) => {
    this.modifyStateAndURL(point, index => index[point.url] = true)
  }

  handleHideEvidence = (point) => {
    this.modifyStateAndURL(point, index => delete index[point.url])
  }

  renderPoint = (point) => {
    if (this.isPointExpanded(point)) {
      return <PointCard key={point.url} url={point.url} point={point} expanded={true}
                        parentPoint={this.parentPoint}
                        onDelete={this.props.onDelete} onCollapse={() => this.handleHideEvidence(point)}/>
    } else {
      return <PointCard key={point.url} point={point} url={point.url} expanded={false}
                        parentPoint={this.parentPoint}
                        onDelete={this.props.onDelete} onExpand={() => this.handleSeeEvidence(point)}/>
    }
  }

  renderEdge = (edge) => {
    if (this.isPointExpanded(edge.node)) {
      return <PointCard key={edge.node.url} point={edge.node} url={edge.node.url} expanded={true}
                        link={edge.link} parentPoint={this.parentPoint}
                        onDelete={this.props.onDelete} onCollapse={() => this.handleHideEvidence(edge.node)}/>
    } else {
      return <PointCard key={edge.node.url} point={edge.node} url={edge.node.url} expanded={false}
                        link={edge.link} parentPoint={this.parentPoint}
                        onDelete={this.props.onDelete} onExpand={() => this.handleSeeEvidence(edge.node)}/>
    }
  }

  renderPoints = (points) => this.props.points.map((point, i) => this.renderPoint(point))

  renderInfinitePoints = (points) => (
    <div className="hideScrollbar">
      <InfiniteScroll next={this.props.loadMorePoints} hasMore={this.props.hasMore}
                      loader={<Spinner />} endMessage={<span className="pointListEndMessage">End of list</span>}>
        {this.renderPoints(points)}
      </InfiniteScroll>
    </div>
  )

  renderPointsWithMoreLink = (points) => <div>
    {this.renderPoints(points)}
    <a onClick={this.props.loadMorePoints}>Load More...</a>
  </div>

  renderEdges = (edges) => edges.map((edge, i) => this.renderEdge(edge))

  showIrrelevant = () => this.setState({hideIrrelevant: false})

  hideIrrelevant = () => this.setState({hideIrrelevant: true})

  relevanceThreshold = () => this.props.relevanceThreshold

  renderRelevantEdges = () => {
    const threshold = this.relevanceThreshold()
    if (threshold) {
      let relevantEdges = this.props.edges.filter(({link: {relevance}}) => relevance > threshold)
      if (relevantEdges.length != this.props.edges.length) {
        return <div>{this.renderEdges(this.state.hideIrrelevant ? relevantEdges : this.props.edges)}
          {this.state.hideIrrelevant ? <a onClick={this.showIrrelevant}>Show points below relevance threshold</a> :
                                       <a onClick={this.hideIrrelevant}>Hide points below relevance threshold</a>}
        </div>
      } else {
        return this.renderEdges(this.props.edges)
      }
    } else {
      return this.renderEdges(this.props.edges)
    }
  }


  render(){
    if (this.props.point) {
      return this.renderPoint(this.props.point);
    } else if (this.props.points) {
      if (this.props.loadMorePoints) {
        if (this.props.infiniteScroll) {
          return this.renderInfinitePoints(this.props.points)
        } else {
          return this.renderPointsWithMoreLink(this.props.points)
        }
      } else {
        return <div>
          {this.renderPoints(this.props.points)}
        </div>
      }
    } else if (this.props.loading || (this.props.data && this.props.data.loading)) {
      return <div className="spinnerPointList"><Spinner /></div>
    } else if (this.props.data && this.props.data.point) {
      return this.renderPoint(this.props.data.point);
    } else if (this.props.edges) {
      return this.renderRelevantEdges()
    } else {
      return <div>dunno what to do...</div>
    }
  }
}

const PointList = withRouter(PointListComponent);

export {PointList};
export const PointListWithPoint = graphql(GetPoint)(PointList);
