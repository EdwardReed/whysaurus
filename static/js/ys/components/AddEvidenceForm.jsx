import React from 'react'
import PropTypes from 'prop-types'
import { Form, Text } from 'react-form';
import { graphql } from 'react-apollo';

import * as validations from '../validations';
import * as schema from '../schema';
import TitleText from './TitleText'
import ClaimSearch from './ClaimSearch'

// use onmousedown here to try to get in before blur hides the UI (see note in TitleText onBlur below)
// TODO: think about ways to make the "suggestion UI hide" condition be "clicking on anything that is not the text input or suggestion ui itself"
const ExistingClaimPicker = ({claims, onSelectClaim}) => <ul>
      {claims && claims.map((claim) => <li onMouseDown={e => onSelectClaim(claim, e)} key={claim.id}>
                            {claim.title}
                            </li>)}
      </ul>

class AddEvidenceForm extends React.Component {

  constructor(props) {
    super(props)
    this.currentSupportingClaimURLs = new Set(this.props.currentSupportingClaims.map(claim => claim.url))
  }

  static propTypes = {
    addExistingClaim: PropTypes.func.isRequired,
    onSubmit: PropTypes.func.isRequired,
    onCancel: PropTypes.func.isRequired
  }

  state = {titleTextFocused: false}

  selectExistingClaim = (claim) =>
    this.props.addExistingClaim(this.props.evidenceType, claim)

  filterCurrentSupport = (claims) =>
    this.props.currentSupportingClaims ? claims.filter(claim => !this.currentSupportingClaimURLs.has(claim.url)) : claims

  existingClaimPicker = (titleValue, searchResults, searching) => {
    if (this.state.titleTextFocused) {
      if (titleValue && (titleValue != '')){
        if (searching) {
          return <div>Searching...</div>
        } else {
          return <ExistingClaimPicker claims={this.filterCurrentSupport(searchResults)} onSelectClaim={this.selectExistingClaim}/>
        }
      } else if (this.props.user) {
        return <ExistingClaimPicker claims={this.filterCurrentSupport(this.props.user.recentlyViewed)} onSelectClaim={this.selectExistingClaim}/>
      }
    }
  }
  
  // TODO: add multiple options for each type and randomize!
  generatePlaceholderText(evidenceType) {
    let placeholderSupport = `Make a claim, eg "Dogs can learn more tricks than cats."`
    let placeholderCounter= `Make a claim, eg "Cats are better than dogs at killing mice."`
    if (this.props.evidenceType=="counter") {
        return placeholderCounter 
    } else {
        return placeholderSupport
    }
  } 

  render(){
    const {userLoading, user} = this.props
    let submitClasses = `buttonUX2 addEvidenceFormButton ${this.props.evidenceType=="counter" ? "buttonUX2Red" : ""}`
    return <Form onSubmit={this.props.onSubmit}
                 validate={values => ({title: validations.validateTitle(values.title)})}>
      { ({submitForm, values: {title}}) => (
        <ClaimSearch
          query={title}
          render={({results, searching}) =>
                  <form onSubmit={submitForm} className="addEvidenceForm">
                      <TitleText id="title" className="titleTextField"
                                   autoComplete='off'
                                   placeholder={this.generatePlaceholderText(this.props.evidenceType)}
                                   onFocus={() => {this.setState({titleTextFocused: true})}}
                                   // use the setTimeout here to allow the mousedown event in existingclaimpicker to fire consistently
                                   // right now this fires before the onClick in ExistingClaimPIcker and hides that UI before the click event can be fired
                                   // TODO: think about ways to make the "suggestion UI hide" condition be "clicking on anything that is not the text input or suggestion ui itself"
                                   onBlur={() => {setTimeout(() => this.setState({titleTextFocused: false}), 100)}}
                          />
                      {this.existingClaimPicker(title, results, searching)}
                      <button type="submit" className={submitClasses}>Add</button>
                      <button type="cancel" className="cancelButton cancelButtonAddEvidence" onClick={this.props.onCancel}>Cancel</button>
                    </form>
                  }/>
      )}
    </Form>
  }
}

export default graphql(schema.CurrentUserQuery, {
  props: ({ownProps, data: {loading, currentUser, refetch}}) => ({
    userLoading: loading,
    user: currentUser,
    refetchUser: refetch
  })
})(AddEvidenceForm)
