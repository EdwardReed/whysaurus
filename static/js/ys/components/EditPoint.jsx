import React from 'react'
import * as validations from '../validations';
import { Form, Text } from 'react-form';
import * as schema from '../schema';
import { graphql, compose } from 'react-apollo';
import TitleText from './TitleText'

const EditTitleForm = ( props ) => {
  return (
    <Form onSubmit={props.onSubmit}
          defaultValues={{title: props.point.title}}
          validate={values => ({title: validations.validateTitle(values.title)})}>
      { formApi => (
        <form onSubmit={formApi.submitForm} id="form1" className="editPointTextForm">
          <TitleText onClick={props.onClick} id="editPointTextField"/>
          <button onClick={props.onClick} className="buttonUX2" type="submit">Save</button>
        </form>
      )}
    </Form>
  );
}


class EditPointComponent extends React.Component {
  constructor(props) {
    super(props)
    this.state = {saving: false}
    this.handleClickSave = this.handleClickSave.bind(this);
    this.handleClickNoProp = this.handleClickNoProp.bind(this);
  }

  // TODO: this simple function is also defined in the PointCard component - can/should it be declared in a single place somehow?
  handleClickNoProp(e) {
    e.stopPropagation();
  }

  get point() {
    return this.props.point;
  }

  handleClickSave(values, e, formApi) {
    values.url = this.point.url
    this.setState({saving: true})
    this.props.mutate({
      variables: values
    })
    // this component will be replaced after save, so we don't need to update state
  }

  render(){
    const score = this.point.pointValue;
    if (this.state.saving) {
      return <div><img id="spinnerImage" className="spinnerPointSubmitButtonPosition" src="/static/img/ajax-loader.gif"/>Saving...</div>;
    } else {
      return <div>
        <span>
        <EditTitleForm onClick={this.handleClickNoProp} onSubmit={this.handleClickSave} point={this.point} countedValue={this.point.title}/>
        </span>
        <button onClick={this.props.onCancel} type="cancel" className="cancelButton">Cancel</button>
      </div>;
    }
  }
}

export default compose(
  graphql(schema.EditPointQuery),
)(EditPointComponent)
