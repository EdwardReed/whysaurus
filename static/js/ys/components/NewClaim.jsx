import React from 'react'
import * as validations from '../validations';
import { Form, Text } from 'react-form';
import TitleText from './TitleText'
import ReactFilestack from 'filestack-react';
import { config } from '../config'


export default class NewClaim extends React.Component {
  constructor(props) {
    super(props)
  }

  render(){
    return <Form onSubmit={this.props.onSubmit}
                 validate={values => ({title: validations.validateTitle(values.title)})}>
      { formApi => (
        <form onSubmit={formApi.submitForm} id="form1" className="addEvidenceForm">
          <TitleText id="title" className="titleTextField" placeholder='Make a claim, eg "Dogs can learn more tricks than cats."' />
          <ReactFilestack
            apikey={config.filestack.key}
            buttonText="Upload Image"
            buttonClass="classname"
            options={{storeTo: {location: 's3'}}}
            onSuccess={(response) => console.log(response)}
            />
          <button type="submit">Add</button>
          <button type="cancel" className="cancelButton cancelButtonAddEvidence" onClick={this.props.onCancel}>Cancel</button>
        </form>
      )}
    </Form>
  }
}
