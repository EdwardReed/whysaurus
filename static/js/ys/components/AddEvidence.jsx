import React from 'react'
import PropTypes from 'prop-types'
import { graphql, compose } from 'react-apollo';

import * as schema from '../schema';
import AddEvidenceForm from './AddEvidenceForm'

class AddEvidence extends React.Component {
  state = {addingSupport: false, addingCounter: false }

  get point() {
    return this.props.point;
  }

  get uiType(){
    return this.props.type
  }

  get adding() {
    return this.state.addingSupport || this.state.addingCounter
  }

  handleClickAddEvidenceSupport = (e) => {
    if (this.props.user){
      this.setState({addingSupport: true, addingCounter: false})
    } else {
      $("#loginDialog").modal("show");
    }
  }

  handleClickAddEvidenceCounter = (e) => {
    if (this.props.user){
      this.setState({addingSupport: false, addingCounter: true})
    } else {
      $("#loginDialog").modal("show");
    }
  }

  handleClickCancel = (e) => {
    e.stopPropagation();
    this.setState({addingSupport: false, addingCounter: false})
    console.log("AddEvidenceCard : handleCancel")
  }

  handleClickSave = (evidenceType, values, e, formApi) => {
    let parentURL = this.point.url
    values.parentURL = parentURL
    values.linkType = evidenceType
    this.setState({saving: true})
    this.props.mutate({
      variables: values,
      update: (proxy, { data: { addEvidence: { newEdges } }}) => {
        const data = proxy.readQuery({ query: schema.GetPoint, variables: {url: parentURL} })
        data.point.relevantPoints.edges = data.point.relevantPoints.edges.concat(newEdges.edges)
        let points = data.point[evidenceType == "supporting" ? "supportingPoints" : "counterPoints"]
        points.edges = points.edges.concat(newEdges.edges)
        proxy.writeQuery({ query: schema.GetPoint,
                           variables: {url: parentURL},
                           data: data })
      }
    }).then( res => {
      this.setState({saving: false,
                     addingSupport: false,
                     addingCounter: false})
      console.log(res)
    })
  }

  handleClickSaveSupport = (values, e, formApi) => {
    this.handleClickSave("supporting", values, e, formApi)
  }

  handleClickSaveCounter = (values, e, formApi) => {
    this.handleClickSave("counter", values, e, formApi)
  }

  addExistingClaim = (claim) => {
    console.log("ADDING EXISTING CLAIM")
    console.log(claim)
    console.log("TODO: UNIMPLEMENTED")
  }

  renderAddEvidenceForm = (evidenceType) => {
    console.log("AddEvidenceCard : renderAddEvidenceForm : " + evidenceType)
    let groupClass = `${(this.numSupportingPlusCounter() > 0) && "verticalOffsetForLongEvidenceArrow"}`
    return <span className={groupClass}>
      { this.state.saving ? <span className="addEvidenceFormSaving"><img id="spinnerImage" className="spinnerPointSubmitButtonPosition" src="/static/img/ajax-loader.gif"/>Saving...</span> :
        <AddEvidenceForm evidenceType={evidenceType} onSubmit={evidenceType=="supporting" ? this.handleClickSaveSupport : this.handleClickSaveCounter} onCancel={this.handleClickCancel} addExistingClaim={this.addExistingClaim}/> }
    </span>
  }

  addText = (evidenceType) => {
    switch (evidenceType){
      case "supporting":
        return "Add Evidence For"
      case "counter":
        return "Add Evidence Against"
      default:
        return "Add Evidence"
    }
  }

  numSupportingPlusCounter = () => {
    return (this.point.numSupporting + this.point.numCounter)
  }

  renderAddEvidenceFormBasedOnState = () => {
    if (this.state.addingSupport) {
      return <span>
        {this.renderAddEvidenceForm("supporting")}
      </span>
    } else if (this.state.addingCounter) {
      return <span>
        {this.renderAddEvidenceForm("counter")}
      </span>
    }
    else return (null)
  }

  renderAddEvidenceButton = (evidenceType) => {
    let classesButton = `buttonUX2 ${evidenceType=="counter" ? "buttonUX2Red" : ""} addEvidenceButton`
    let nameButton = `${evidenceType=="counter" ? "addCounterEvidenceButton" : "addSupportingEvidenceButton" }`
    let buttonLabel = this.addText(evidenceType)
    let aClass = `${(this.numSupportingPlusCounter() > 0) && "verticalOffsetForLongEvidenceArrow"}`
    let onClick = evidenceType == 'supporting' ? this.handleClickAddEvidenceSupport : this.handleClickAddEvidenceCounter
    return <a className={aClass} onClick={onClick}>
      <button type="button" name={nameButton} tabIndex="0" className={classesButton}>{buttonLabel}</button>
    </a>
  }

  renderSupportButton = () => this.renderAddEvidenceButton("supporting")

  renderCounterButton = () => this.renderAddEvidenceButton("counter")

  renderDualButtons = () => <span>
    {this.renderSupportButton()}
    {this.renderCounterButton()}
  </span>

  renderEvidenceArrow = (color) => {
    let arrowGrpClass = `arrowEvidence ${(this.numSupportingPlusCounter() > 0) && "verticalOffsetForLongEvidenceArrow"}`
    let arrowHeadClass = `arrowHeadUp ${color == "red" && "arrowHeadUpRed"}`
    let arrowStemClass = `arrowStemEvidence ${(this.numSupportingPlusCounter() == 0) && "arrowStemEvidenceShort" } ${color == "red" && "arrowStemRed"}`
    return <div className={arrowGrpClass}>
      <div className={arrowHeadClass}></div>
      <div className={arrowStemClass}></div>
    </div>
  }

  render() {
    let topDivClass = `addEvidenceUI`
    switch (this.uiType) {
    case "DUAL":
      return <div className={topDivClass}>
        { this.renderEvidenceArrow() }
        { (this.adding) ? this.renderAddEvidenceFormBasedOnState() : this.renderDualButtons() }
      </div>
    case "SUPPORT":
      return <div className={topDivClass}>
        { this.renderEvidenceArrow() }
        { this.state.addingSupport ? this.renderAddEvidenceForm("supporting") : this.renderSupportButton() }
      </div>
    case "COUNTER":
      return <div className={topDivClass}>
        { this.renderEvidenceArrow("red") }
        { this.state.addingCounter ? this.renderAddEvidenceForm("counter") : this.renderCounterButton() }
      </div>
    default:
      console.log("AddEvidenceCard : render() : something's wrong!")
      return <div className={topDivClass}>
      </div>
    }
  }
}


export default compose(
  graphql(schema.AddEvidenceQuery),
  graphql(schema.CurrentUserQuery, {
    props: ({ownProps, data: {loading, currentUser, refetch}}) => ({
      userLoading: loading,
      user: currentUser,
      refetchUser: refetch
    })
  })
)(AddEvidence)
