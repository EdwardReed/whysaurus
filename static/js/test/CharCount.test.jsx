import React from 'react';
import { CharCount } from '../ys/components/CharCount'
import renderer from 'react-test-renderer'
import { Form, Text } from 'react-form';

test('CharCount component renders', () => {
  const component = renderer.create(
    <Form>
      {formApi => (
        <form>
          <CharCount maxChars={200}
                     render={({charsLeft, countedFieldOnChange}) => (
            <div>
              <Text onChange={countedFieldOnChange} />
              <span id="charsLeft">{charsLeft}</span>
            </div>
            )}/>
        </form>
      )}
    </Form>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

     //TODO: add enzyme and do real dom testing
});
