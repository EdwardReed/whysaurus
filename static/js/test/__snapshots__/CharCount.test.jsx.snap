// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharCount component renders 1`] = `
<form>
  <div>
    <input
      onBlur={[Function]}
      onChange={[Function]}
      value={Object {}}
    />
    <span
      id="charsLeft"
    >
      200
    </span>
  </div>
</form>
`;
